app/models/mixins/inter_region_api_method_relay.rb

```diff
@@ -70,8 +70,11 @@ def self.api_client_connection_for_region(region_number, user = User.current_use
 
   def self.exec_api_call(region, collection_name, action, api_args = nil, id = nil)
     require 'manageiq-api-client'
+    _log.info("Executing InterRegion api call for region #{region} with arguments: #{api_args}")
     api_args ||= {}
-    collection = api_client_connection_for_region(region).public_send(collection_name)
+    user_name =  api_args[:requester]&.fetch("user_name")
+    user_name = User.current_user&.userid if user_name.blank?
```

Comment:
Why the `&.` here?

Reply:
Retirement request now going throw queue, so, ```User.current_user``` (or ```User,curent_userid```) is nil

Reply:
for retirement request we need ```&```

Reply:
this fix does does make  sense  right now only for retirement request, but going forward all operations toward remote region over queue can use it